packages/web-app-files/src/views/LocationPicker.vue

```diff
@@ -145,11 +148,28 @@ export default {
           )
           break
         default:
-          resources = yield ref.$client.files.list(
-            buildWebDavFilesPath(ref.user.id, target),
-            1,
-            DavProperties.Default
-          )
+          if (unref(hasShareJail)) {
+            const graphClient = clientService.graphAuthenticated(
+              ref.$store.getters.configuration.server,
+              ref.$store.getters.getToken
+            )
+            const userResponse = yield graphClient.users.getMe()
+            if (!userResponse.data) {
+              throw new Error('graph.user.getMe() has no data')
+            }
+
+            resources = yield ref.$client.files.list(
+              buildWebDavSpacesPath(userResponse.data.id, target),
+              1,
+              DavProperties.Default
+            )
+          } else {
+            resources = yield ref.$client.files.list(
+              buildWebDavFilesPath(ref.user.id, target),
+              1,
+              DavProperties.Default
+            )
+          }
```

Comment:
Since only the user ID differs, it may be nicer to save that in a variable and then call `yield ref.$client.files.list()` outside the if-block.

Reply:
👍